### DISCUSSION TITLE
Random digits added to imported datapoints

### DISCUSSION THREAD

**Discussion by moPellowe** [Open-Systems-Pharmacology/OSPSuite.Core#1601] [General]
I've noticed that random digits get added to imported data points with decimals in PK-Sim.

I imported an excel sheet (attached below) into PK-Sim for a generic compound (I just loaded the Aciclovir template), and when I look at the data within the building block, there are digits added that weren't in the original xlsx file.

![image](https://user-images.githubusercontent.com/57540473/164476337-296d43b8-1902-4367-95f8-b53e718ecd26.png)
![image](https://user-images.githubusercontent.com/57540473/164476408-a8f35ef7-5c14-40a7-9f49-912c703d2df2.png)

[test.xlsx](https://github.com/Open-Systems-Pharmacology/Forum/files/8531646/test.xlsx)

**Comment by msevestre** [comment]
@moPellowe Transferring this to our code repository.
I believe this is a display issue only when converting values from float to double

 @Yuri05 The problem happens here I believe
https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/blob/develop/src/OSPSuite.Infrastructure.Export/DataRepositoryExportTask.cs#L178

We are converting a float to double via the UnitConversion methods. Then the DataTable is bound to the grid for display and the value has been corrupted 

The same happens when we are loading the value and saving it from display unit in the file to internal unit in the column

I feel that we should create an overload that converts a float to float for unit conversion. Internally, we should only use float, also in the DataTable

**Reply by Yuri05** [reply]
>I feel that we should create an overload that converts a float to float for unit conversion. Internally, we should only use float, also in the DataTable

Agree.
